### PR TITLE
New rule: open_redirect_weblinkconnect.yml

### PR DESCRIPTION
**Pull request by aidenmitchell**

### CODE REVIEW

**Comment by aidenmitchell** [comment]
No hits

File: detection-rules/open_redirect_weblinkconnect.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          .href_url.domain.domain == 'api-internal.weblinkconnect.com'
+          and strings.icontains(.href_url.path, "/click")
+          and strings.icontains(.href_url.query_params, "url=")
+  )
+  and sender.email.domain.root_domain != 'weblinkconnect.com'
+
```
Comment (zoomequipd) [inline detection-rules/open_redirect_weblinkconnect.yml:13]: include high trust sender negation? 

```suggestion
  // negate highly trusted sender domains unless they fail DMARC authentication
  and (
    (
      sender.email.domain.root_domain in $high_trust_sender_root_domains
      and not headers.auth_summary.dmarc.pass
    )
    or sender.email.domain.root_domain not in $high_trust_sender_root_domains
  )
```

**Review by zoomequipd — APPROVED** [review]

**Pull request merged**